### ISSUE TITLE
USE ... AS

### ISSUE THREAD

**Issue by thekid** [tueftler/doget#12]
## Usecase

Given this `Dockerfile.in`:

```dockerfile
FROM registry.example.com/base/debian:jessie

# Selfbuilt JRE including example.com CA certificates
USE registry.example.com/docker/openjdk/8-jre

# Neo4J as-is. Use Commit SHA to pin version we've audited
USE github.com/neo4j/docker-neo4j-publish/3.0.4/community:1ad894c
```

This yields the error Include github.com/neo4j/docker-neo4j-publish/3.0.4/community:master inherits from java:openjdk-8-jre, which is incompatible with registry.example.com/base/debian:jessie and exits. 

My first idea was to [add a -force flag](https://github.com/tueftler/doget/pull/11). The problems however are it forces *all* conflicts to be resolved. During development, authors will have a careful oversight of the messages. However, once an automated build is set up, and stuff is updated and refactored over the time, problems will begin to sneak in.

## Solution
In a nutshell:

```dockerfile
FROM registry.example.com/base/debian:jessie

USE registry.example.com/docker/openjdk/8-jre AS java:openjdk-8-jre
#                                             ^^^^^^^^^^^^^^^^^^^^^
#                                             This

USE github.com/neo4j/docker-neo4j-publish/3.0.4/community:1ad894c
```

The *FROM* instruction would most probably also need to have an `AS` extension.

**Comment by thekid** [comment]
In the case that we are the trait's authors, we could also have the following embedded *inside* the trait.

```dockerfile
FROM registry.example.com/base/debian:jessie

PROVIDES php:5.6-cli php:5-cli

RUN apt-get update && apt-get install -y \
    php5-json \
    php5-cli \
# ...
  && rm -rf /var/lib/apt/lists/*
```

The *PROVIDES* instruction would provide a list of docker labels this trait is compatible with.

**Comment by thekid** [comment]
After some thought, we don't really need  AS :

```dockerfile
USE github.com/docker-library/php/7.1 AS php:latest

# ...is the same as:
USE github.com/docker-library/php/7.1
PROVIDES php:latest
```

@mikey179 what do you think?

**Comment by mikey179** [comment]
I think it's great, reducing complexity of the feature. Also, it allows for more than one image compared to the AS approach, and I think it will be possible to prevent the problems I raised in #14 with my USE ... AS implementation.
